pkg/knuu/knuu.go

```diff
@@ -39,24 +39,23 @@ type Knuu struct {
 }
 
 type Options struct {
-	K8s          k8s.KubeManager
 	TestScope    string
 	ImageBuilder builder.Builder
-	Minio        *minio.Minio
-	Timeout      time.Duration
+	MinioClient  *minio.Minio
 	ProxyEnabled bool
+	Timeout      time.Duration
 	Logger       *logrus.Logger
 }
 
-func New(ctx context.Context, opts Options) (*Knuu, error) {
+func New(ctx context.Context, k8sClient k8s.KubeManager, opts Options) (*Knuu, error) {
```

Comment:
why did you pull this out of the options? we should just leave it in the options.

Reply:
because this is mandatory to have and it is not an optional parameter.
If we want to have Minio and probably some other services in future initiated by the user, which is a good practice, then we need to  use the same `k8sClient` for both. For example if we use one `k8sClient` for Minio and `Knuu` also makes its own, we will have two `k8sClinets` which introduces inconsistency also it might point to a different namespaces. So we have to extract it and ask user to pass it in.